### PR TITLE
add session status updating

### PR DESCRIPTION
**Pull request by failandimprove1**
sets status of session to running when creating it, sets it to finished when the a.initiate.chat function is done running

### CODE REVIEW

File: apps/api/src/models/session.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+from enum import StrEnum, auto
 from uuid import UUID, uuid4
 
 from pydantic import BaseModel, Field
 
 
+class SessionStatus(StrEnum):
+    RUNNING     = auto()
+    FINISHED    = auto()
+    CANCELLED   = auto()
```
Comment (eksno) [inline apps/api/src/models/session.py]: I think using a pydantic model with Literal["running"] is better here for db compatibility. Should be part of the session model

File: apps/api/src/tooling/langchain_tooling.py [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
+
+    #@classmethod
+    #def from_string(cls, value):
+    #    try:
+    #        formatted_string = cls[value.upper().replace("-", "_")]
+    #        logger.warning("cool string: %s" % formatted_string)
+    #        return formatted_string
+    #    except KeyError:
+    #        logger.warning("yo what the hell")
+    #        return cls.INVALID
```
Comment (eksno) [inline apps/api/src/tooling/langchain_tooling.py]: nit: avoid commented out code

File: apps/api/src/tooling/langchain_tooling.py [hunk 3]
```diff
@@ -1,30 +1,43 @@
...
 
     INVALID = ""
 
 
 def generate_llm_config(tools: list[BaseTool]) -> list[dict]:
+    """
+    Generate a valid function schema for the autogen llm config from the given tool.
+
+    
+    """
```
Comment (eksno) [inline apps/api/src/tooling/langchain_tooling.py]: there's empty lines here, remember to end off with a format commit

**Review by eksno — DISMISSED** [review]
See comments

File: apps/api/src/models/session.py (hunk 1 above)
Comment (failandimprove1) [reply apps/api/src/models/session.py]: The SessionStatus property "RUNNING" is interpreted as the string "running" by the interpreter during "compile time" according to [the documentation](https://docs.python.org/3/library/enum.html#enum.StrEnum). If the db compatibility is a concern we could remove auto and replace with a literal, but I think an enum is easier to maintain. 
[in this function](https://github.com/Futino/aitino/blob/e44e543f832d3331e8a79af9b35ec069dee942a4/apps/api/src/interfaces/db.py#L114)

**Pull request merged**